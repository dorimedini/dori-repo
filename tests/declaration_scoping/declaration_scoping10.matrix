{
	int x=1;
	{
		print "Should be integer 1";
		print x;
		int x;
		{
			{
				print "Should be integer 0";
				print x;
				matrix x(1,1) = [2];
				{
			        	print "Should be matrix [2]";
					print x;
					{
						int x = 3;
						{
							print "Should be integer 3";
							print x;
						}
					}
					print "Should be matrix [2]";
					print x;
				}
			}
			print "Should be integer 0";
			print x;
		}
	        print "Should be integer 0"; 
                print x;
	}
	print "Should be integer 1"; 
        print x;
}
