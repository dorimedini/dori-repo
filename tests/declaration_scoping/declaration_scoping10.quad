s[0]=1
printLn "Should be integer 1"
printLn s[0]
s[1]=0
printLn "Should be integer 0"
printLn s[1]
s[2]=2
s[3]=s[2]
printLn "Should be matrix [2]"
print "["
print s[3]
printLn "]"
s[4]=3
printLn "Should be integer 3"
printLn s[4]
printLn "Should be matrix [2]"
print "["
print s[3]
printLn "]"
printLn "Should be integer 0"
printLn s[1]
printLn "Should be integer 0"
printLn s[1]
printLn "Should be integer 1"
printLn s[0]
