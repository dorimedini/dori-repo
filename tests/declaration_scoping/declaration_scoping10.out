Should be integer 1
1
Should be integer 0
0
Should be matrix [2]
[2]
Should be integer 3
3
Should be matrix [2]
[2]
Should be integer 0
0
Should be integer 0
0
Should be integer 1
1
